coderd/database/dbfake/dbfake.go

```diff
@@ -161,42 +164,18 @@ func (b WorkspaceBuildBuilder) Do() database.WorkspaceBuild {
 	})
 	require.NoError(b.t, err, "complete job")
 
-	// This intentionally fulfills the minimum requirements of the schema.
-	// Tests can provide a custom version ID if necessary.
-	if b.seed.TemplateVersionID == uuid.Nil {
-		jobID := uuid.New()
-		templateVersion := dbgen.TemplateVersion(b.t, b.db, database.TemplateVersion{
-			JobID:          jobID,
-			OrganizationID: b.ws.OrganizationID,
-			CreatedBy:      b.ws.OwnerID,
-			TemplateID: uuid.NullUUID{
-				UUID:  b.ws.TemplateID,
-				Valid: true,
-			},
-		})
-		payload, _ := json.Marshal(provisionerdserver.TemplateVersionImportJob{
-			TemplateVersionID: templateVersion.ID,
-		})
-		dbgen.ProvisionerJob(b.t, b.db, nil, database.ProvisionerJob{
-			ID:             jobID,
-			OrganizationID: b.ws.OrganizationID,
-			Input:          payload,
-			Type:           database.ProvisionerJobTypeTemplateVersionImport,
-			CompletedAt: sql.NullTime{
-				Time:  dbtime.Now(),
-				Valid: true,
-			},
-		})
-		ProvisionerJobResources(b.t, b.db, jobID, b.seed.Transition, b.resources...).Do()
-		b.seed.TemplateVersionID = templateVersion.ID
-	}
-	build := dbgen.WorkspaceBuild(b.t, b.db, b.seed)
+	resp.Build = dbgen.WorkspaceBuild(b.t, b.db, b.seed)
 	ProvisionerJobResources(b.t, b.db, job.ID, b.seed.Transition, b.resources...).Do()
 	if b.ps != nil {
-		err = b.ps.Publish(codersdk.WorkspaceNotifyChannel(build.WorkspaceID), []byte{})
+		err = b.ps.Publish(codersdk.WorkspaceNotifyChannel(resp.Build.WorkspaceID), []byte{})
 		require.NoError(b.t, err)
 	}
-	return build
+
+	for i := range b.params {
+		b.params[i].WorkspaceBuildID = resp.Build.ID
+	}
+	_ = dbgen.WorkspaceBuildParameters(b.t, b.db, b.params)
```

Comment:
I think this should happen before `Publish()` to the pubsub, to more closely match what provionerd would do; e.g. some subscribers might expect the parameters to have been set, and would race against your insert here.